framework/src/application/node/forger/forger.ts

```diff
@@ -12,28 +12,104 @@
  * Removal or modification of this copyright notice is prohibited.
  */
 
-'use strict';
-
-const {
+import {
 	getPrivateAndPublicKeyBytesFromPassphrase,
 	decryptPassphraseWithPassword,
 	parseEncryptedPassphrase,
 	hashOnion,
 	generateHashOnionSeed,
 	getAddressFromPublicKey,
-} = require('@liskhq/lisk-cryptography');
-
-const {
+} from '@liskhq/lisk-cryptography';
+import { Chain, BlockInstance } from '@liskhq/lisk-chain';
+import { Dpos } from '@liskhq/lisk-dpos';
+import { BFT } from '@liskhq/lisk-bft';
+import { BaseTransaction } from '@liskhq/lisk-transactions';
+import { TransactionPool } from '@liskhq/lisk-transaction-pool';
+import {
 	FORGER_INFO_KEY_USED_HASH_ONION,
 	FORGER_INFO_KEY_REGISTERED_HASH_ONION_SEEDS,
-} = require('./constant');
-const { HighFeeForgingStrategy } = require('./strategies');
+} from './constant';
+import { HighFeeForgingStrategy } from './strategies';
+import { Logger, Storage, StringKeyVal } from '../../../types';
+
+interface UsedHashOnion {
+	readonly count: number;
+	readonly address: string;
+	readonly height: number;
+}
+
+interface DelegateConfig {
+	readonly publicKey: string;
+	readonly encryptedPassphrase: string;
+	readonly hashOnion: HashOnionConfig;
+}
+
+interface HashOnionConfig {
+	readonly count: number;
+	readonly distance: number;
+	readonly hashes: string[];
+}
+
+interface KeyPair {
+	[key: string]: Buffer;
+}
 
-class Forger {
-	constructor({
+interface KeyPairs {
+	[key: string]: KeyPair;
+}
+
+interface ProcessorModule {
+	readonly create: (input: {
+		readonly keypair: KeyPair;
+		readonly timestamp: number;
+		transactions: BaseTransaction[];
+		readonly previousBlock: BlockInstance;
+		readonly seedReveal: string;
+	}) => Promise<BlockInstance>;
+	readonly process: (block: BlockInstance) => Promise<void>;
+}
+
+interface ForgerConstructor {
+	readonly forgingStrategy?: HighFeeForgingStrategy;
+	readonly logger: Logger;
+	readonly storage: Storage;
+	readonly processorModule: ProcessorModule;
+	readonly dposModule: Dpos;
+	readonly bftModule: BFT;
+	readonly transactionPoolModule: TransactionPool;
+	readonly chainModule: Chain;
+	readonly maxPayloadLength: number;
+	readonly forgingDelegates?: ReadonlyArray<DelegateConfig>;
+	readonly forgingForce?: boolean;
+	readonly forgingDefaultPassword?: string;
+	readonly forgingWaitThreshold: number;
+}
+
+export class Forger {
+	private readonly logger: Logger;
```

Comment:
can we add underscore as a prefix to private properties as some functions in the same class are following that pattern so it looks a bit inconsistent